src/lib/project-detection/index.ts

```diff
@@ -123,13 +125,35 @@ export const getJobNumberFromURL = (url: string): number | undefined => {
 export const getProjectSlugFromURL = (url: string) => {
   const urlWithoutQuery = url.split('?')[0];
   const parts = urlWithoutQuery.split('/');
+
+  let startIndex = -1;
   const pipelineIndex = parts.indexOf('pipelines');
-  if (pipelineIndex === -1) {
-    throw new Error('Invalid CircleCI URL format');
+  if (pipelineIndex !== -1) {
+    startIndex = pipelineIndex + 1;
+  } else {
+    for (const vcs of vcses) {
+      const shortIndex = parts.indexOf(vcs.short);
+      const nameIndex = parts.indexOf(vcs.name);
+      if (shortIndex !== -1) {
+        startIndex = shortIndex;
+        break;
+      }
+      if (nameIndex !== -1) {
+        startIndex = nameIndex;
+        break;
+      }
+    }
   }
-  const vcs = parts[pipelineIndex + 1];
-  const org = parts[pipelineIndex + 2];
-  const project = parts[pipelineIndex + 3];
+
+  if (startIndex === -1) {
+    throw new Error(
+      'Error getting project slug from URL: Invalid CircleCI URL format',
+    );
+  }
+
+  const vcs = parts[startIndex];
+  const org = parts[startIndex + 1];
+  const project = parts[startIndex + 2];
```

Comment:
Another option here...

```typescript
const PROJECT_SLUG_PARTS = 3; // vcs/org/project
```

```suggestion
  const [vcs, org, project] = parts.slice(
    startIndex,
    startIndex + PROJECT_SLUG_PARTS,
  );
```